src/tree-sitter-language-mode.js

```diff
@@ -1010,7 +1010,8 @@ class HighlightIterator {
       if (
         next.offset === first.offset &&
         next.atEnd === first.atEnd &&
-        next.depth > first.depth
+        next.depth > first.depth &&
+        next.openTags.length + next.closeTags.length > 0
```

Comment:
Oh wow you ended up with a really tiny diff! I think this fix makes sense.

Down the road, if this type of problem occurs in other cases, we'll probably just have to make this conditional even more specific.